src/modules/Game.class.js

```diff
@@ -0,0 +1,254 @@
+'use strict';
+
+/**
+ * This class represents the game.
+ * Now it has a basic structure, that is needed for testing.
+ * Feel free to add more props and methods if needed.
+ */
+class Game {
+  static get deafultSize() {
+    return 4;
+  }
+
+  static get deafultInitialState() {
+    return new Array(Game.deafultSize)
+      .fill(null)
+      .map((el) => new Array(Game.deafultSize).fill(0));
+  }
+  // current game status
+  #status = 'idle';
+  // current score
+  #score = 0;
+  // flag for Game objects exiting only for checking moves
+  #isForCheck = false;
+  // current state of gamefield
+  #state;
+  // initial state of gamefield
+  #initialState;
+  /**
+   * Creates a new game instance.
+   *
+   * @param {number[][]} initialState
+   * The initial state of the board.
+   * @default
+   * [[0, 0, 0, 0],
+   *  [0, 0, 0, 0],
+   *  [0, 0, 0, 0],
+   *  [0, 0, 0, 0]]
+   *
+   * If passed, the board will be initialized with the provided
+   * initial state.
+   */
+  constructor(initialState = Game.deafultInitialState, isForCheck = false) {
+    this.#initialState = initialState.map((row) => [...row]);
+    this.#applyInitialState();
+    this.#isForCheck = isForCheck;
+  }
+
+  moveLeft() {
+    let isStateChanged = false;
+
+    if (this.#status !== 'playing' && !this.#isForCheck) {
+      return isStateChanged;
+    }
+
+    this.#state.forEach((row) => {
+      const mergedCellsIndexes = new Set();
+      const emptyIndexes = [];
+
+      for (let i = 0; i < row.length; i++) {
+        if (row[i] === 0) {
+          emptyIndexes.push(i);
+          continue;
+        }
+
+        let cellIndex = i;
+
+        if (emptyIndexes.length) {
+          cellIndex = emptyIndexes.shift();
+          emptyIndexes.push(i);
+
+          row[cellIndex] = row[i];
+          row[i] = 0;
+          isStateChanged = true;
+        }
+
+        const previousCellIndex = cellIndex - 1;
+
+        if (
+          previousCellIndex >= 0 &&
+          row[previousCellIndex] === row[cellIndex] &&
+          !mergedCellsIndexes.has(previousCellIndex)
+        ) {
+          mergedCellsIndexes.add(previousCellIndex);
+          row[previousCellIndex] *= 2;
+          row[cellIndex] = 0;
+          emptyIndexes.unshift(cellIndex);
+          this.#score += row[previousCellIndex];
+          isStateChanged = true;
+
+          if (row[previousCellIndex] >= 2048) {
+            this.#status = 'win';
+          }
+        }
+      }
+    });
+
+    if (isStateChanged) {
+      this.#addNumber();
+    }
+
+    if (!this.#isForCheck && !this.#isMovePossible()) {
```

Comment:
The method '#isMovePossible' is checking if any moves are possible after a move has been made. This could be refactored to avoid creating a new Game instance every time, which is costly in terms of performance.